src/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/MemoryMappedFileBlock.cs

```diff
@@ -24,22 +25,22 @@ internal unsafe MemoryMappedFileBlock(IDisposable accessor, SafeBuffer safeBuffe
 
         ~MemoryMappedFileBlock()
         {
-            Dispose(false);
+            DisposeImpl();
         }
 
-        protected override void Dispose(bool disposing)
+        public override void Dispose()
         {
-            if (_safeBuffer != null)
-            {
-                _safeBuffer.ReleasePointer();
-                _safeBuffer = null;
-            }
-
-            if (_accessor != null)
-            {
-                _accessor.Dispose();
-                _accessor = null;
-            }
+            // Suppress finalization _before_ we start disposing.
+            // Otherwise the disposal code could be interrupted and reentered by finalization thread when finalization is triggered on shutdown.
```

Comment:
This can result in memory leaks. `GC.SuppressFinalize` should be done after the cleanup.